### PR TITLE
Tree: add support for Path based Anchors in object forest

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Replace the hacky untested object anchors in object-forest with proper path based one.

Unify forest anchor types with tree anchor types.

Add test for anchors.

### CODE REVIEW

File: packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts [hunk 1]
```diff
@@ -209,55 +214,43 @@ export class ObjectForest extends SimpleDependee implements IEditableForest {
...
-                if (this.search(node, cursorToMove) === TreeNavigationResult.NotFound) {
-                    cursorToMove.clear();
-                } else {
-                    return TreeNavigationResult.Ok;
-                }
-            }
-        }
-
-        return TreeNavigationResult.NotFound;
+       this.getAtPath(path, cursorToMove, observer);
```
Comment (alex-pardes) [inline packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts:223]:
```suggestion
        this.getAtPath(path, cursorToMove, observer);
```

File: packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts [hunk 2]
```diff
@@ -209,55 +214,43 @@ export class ObjectForest extends SimpleDependee implements IEditableForest {
...
-    private search(destination: JsonableTree, cursor: Cursor): TreeNavigationResult {
-        if (cursor.getNode() === destination) {
-            return TreeNavigationResult.Ok;
-        }
+    /**
+     * Set `cursorToMove` to location described by path.
+     * This is NOT a relative move: current position is discarded.
+     * Path must point to existing node.
+     */
+    getAtPath(destination: UpPath, cursorToMove: ITreeSubscriptionCursor, observer?: ObservingDependent): void {
```
Comment (alex-pardes) [inline packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts:232]: I would consider renaming this to something like `moveCursorTo`, `placeCursor`, or `setCursorPosition`, and similarly renaming the other `get` flavor(s).

**Review by alex-pardes — APPROVED** [review]

File: packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts (hunk 2 above)
Comment (CraigMacomber) [reply packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts]: done

File: api-report/tree.api.md [hunk 3]
```diff
@@ -226,11 +221,11 @@ export interface IEditableForest extends IForestSubscription {
 // @public
 export interface IForestSubscription extends Dependee {
     allocateCursor(): ITreeSubscriptionCursor;
-    root(range: DetachedField): ForestAnchor;
+    root(range: DetachedField): Anchor;
     // (undocumented)
     readonly rootField: DetachedField;
     readonly schema: SchemaRepository & Dependee;
-    tryGet(destination: ForestAnchor, cursorToMove: ITreeSubscriptionCursor, observer?: ObservingDependent): TreeNavigationResult;
+    tryMoveCursorTo(destination: Anchor, cursorToMove: ITreeSubscriptionCursor, observer?: ObservingDependent): TreeNavigationResult;
```
Comment (yann-achard-MS) [inline api-report/tree.api.md:228]: This doesn't need to be addressed in this PR, but why can't the `tryMoveCursor` operation be exposed on the cursor to be moved?

Comment (CraigMacomber) [reply api-report/tree.api.md:228]: which things are on forest vs the cursor are pretty arbitrary. We should probably revisit that when we try and do documentation for the observation tracking, and see if we can structure things in a way thats more clear.

**Pull request merged**